interfaces/builtin/docker_support.go

```diff
@@ -647,7 +807,83 @@ const dockerSupportPrivilegedAppArmor = `
 # These rules are here to allow Docker to launch unconfined containers but
 # allow the docker daemon itself to go unconfined. Since it runs as root, this
 # grants device ownership.
-change_profile unsafe /**,
+# TODO: When we drop support for executing other snaps from devmode snaps (or 
+# when the AppArmor parser bugs are fixed) this can go back to the much simpler
+# rule:
+# change_profile unsafe /**,
+# but until then we need this set of rules to avoid exec transition conflicts.
+# See also the comment above the "change_profile unsafe /** -> docker-default," 
+# rule for more context.
+change_profile unsafe /[^s]**,
+change_profile unsafe /s[^n]**,
+change_profile unsafe /sn[^a]**,
+change_profile unsafe /sna[^p]**,
+change_profile unsafe /snap[^/]**,
+change_profile unsafe /snap/{s[^n],c[^o]}**,
```

Comment:
As above - do we need a line above this of the form:

```
 change_profile unsafe /snap/{[^s],[^c]}**,
```

Reply:
done